pkg/binding/definition.go

```diff
@@ -215,14 +215,20 @@ var _ Definition = (*stringOfMapDefinition)(nil)
 func (d *stringOfMapDefinition) GetPath() []string { return d.path }
 
 func (d *stringOfMapDefinition) Apply(u *unstructured.Unstructured) (Value, error) {
-	val, ok, err := unstructured.NestedFieldNoCopy(u.Object, d.path...)
+	strval, ok, err := getValuesByJSONPath(u.Object, d.path...)
 	if err != nil {
 		return nil, err
 	}
 	if !ok {
 		return nil, errors.New("not found")
 	}
 
+	var val interface{}
+	err = json.Unmarshal([]byte(strval), &val)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Instead of repeating this block 3 times in the code, how about that `getValuesByJSONPath` returns `[]interface` directly? Then we could make some assertions based on the expected outcome (e..g if just one string should be selected, then we should expect just single entry, etc.)?